### DISCUSSION TITLE
Add option to generate SQL based on `schema.rs`

### DISCUSSION THREAD

**Discussion by henryboisdequin** [diesel-rs/diesel#2651] [Ideas]
As Diesel generates the `schema.rs` file from SQL, it would be great to have an option to do the reverse. I could define my tables inside `schema.rs` and run a command to generate the SQL. As many full-stack developers have never written SQL, having this option would make it easier to get started with Diesel. This is inspired by Typeorm which generates SQL in the background from Typescript classes instead of generating Typescript classes from SQL.

**Comment by weiznich** [comment]
I definitively can understand why people may want to have such a feature. The point is it is much more complicated to get that really right as it seem at the first look. Additionally I think it's currently not possible to encode some constraints (like unique constraints and so on, or general indices) as part of a `schema.rs` file. The point is as soon as you add all this features we are getting really close to the complexity of SQL itself, so I would say that the benefit of having something like this is quite low.
That all written: This does not depend on diesels internals and is likely a good point to experiment with. This can be done by any third party crate by providing a module wide proc macro applied to the `schema.rs` file or some other external tool that parses this file. Such a crate is then free to experiment and collect some experience if and how such a approach can work. If it has reached some amount of stability we can talk about including it into diesel or promoting this from diesels side.

**Comment by henryboisdequin** [comment]
@weiznich Thanks for your input, I'll be looking into creating a crate such as the one you suggested
